oshmem/mca/spml/base/spml_base.c

```diff
@@ -80,41 +96,97 @@ int mca_spml_base_test(void* addr, int cmp, void* value, int datatype, int *out_
         SPML_BASE_DO_CMP((*out_value), int_addr, cmp, int_value);
         break;
 
+        /* Unsigned Int */
+    case SHMEM_UINT:
+        uint_value = *(unsigned int*) value;
+        uint_addr = (unsigned int*) addr;
+        SPML_BASE_DO_CMP((*out_value), uint_addr, cmp, uint_value);
+        break;
```

Comment:
maybe we can define macro like this
```
#define SPML_BASE_TEST_CASE(_type, _shmem_type, _addr, _value, _cmp, _out_value) \
    case _shmem_type: \
        { \
            _type typed_value = *(const _type*)value; \
            const _type *typed_addr = (const _type*)addr; \
            SPML_BASE_DO_CMP((*_out_value), typed_addr , _cmp, typed_value); \
        } \
        break; \
```

and similarly for wait, wait_until
there is too much code duplication and hard to notice issues..